xds/internal/xdsclient/authority.go

```diff
@@ -399,7 +453,8 @@ func (a *authority) handleWatchTimerExpiry(rType xdsresource.Type, resourceName
 	a.resourcesMu.Lock()
 	defer a.resourcesMu.Unlock()
 
-	if state.wState == watchStateCanceled {
+	if state.wState != watchStateRequested {
```

Comment:
If state is canceled, we shouldn't be doing anything here. So, I think we should handle both states with an early return.

```golang
switch state.wState {
	case watchStateRequested:
		a.logger.Warningf("Unexpected watch state %q for resource %q", state.wState, watchStateRequested)
		return
	case watchStateCanceled:
		return
}
```

Reply:
sure, these early return are more readable, and helps with indentation also while having to line wrap more often for a log statement inside of a if block